src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy

```diff
@@ -88,13 +91,47 @@ class GitVersionPlugin implements Plugin<Project> {
             // first to preserve this behavior in cases where this call would fail but native "git" call does not.
             new DescribeCommand(git.getRepository()).call()
 
-            return GitCli.runGitCommand(project.rootDir, "describe", "--tags", "--always", "--first-parent",
-                    "--match=${prefix}*")
+            /*
+             * Mimick 'git describe --tags --always --first-parent --match=${prefix}*' by using rev-list to
+             * support versions of git < 1.8.4
+             */
+
+            // Get SHAs of all tags, we only need to search for these later on
+            Set<String> tagRefs = Sets.newHashSet()
+            for (String tag : getLines(GitCli.runGitCommand(project.rootDir, "show-ref", "--tags", "-d"))) {
+                List<String> parts = Splitter.on(' ').splitToList(tag)
```

Comment:
pull out Splitter.on() into a static var, same in getLines as well. Perf optimizations =)

Reply:
https://github.com/google/guava/wiki/StringsExplained